deployments/databricks/deploy_collector.sh

```diff
@@ -0,0 +1,188 @@
+#   Directions for creating a PAT: https://docs.databricks.com/en/dev-tools/auth/pat.html
+
+# Optional Variables:
+# - SPLUNK_OTEL_VERSION: Version of the Splunk OpenTelemetry Collector to deploy as a part of this release.
+#   Default: "latest". Valid version must be >=0.119.0.
+# - SCRIPT_DIR: Installation path for the Collector and its config
+# - SPLUNK_REALM: Splunk o11y realm to send data to. Default: us0
+
+DATABRICKS_CLUSTER_HOSTNAME=${DATABRICKS_CLUSTER_HOSTNAME:-}
+SPLUNK_ACCESS_TOKEN=${SPLUNK_ACCESS_TOKEN:-}
+DATABRICKS_ACCESS_TOKEN=${DATABRICKS_ACCESS_TOKEN:-}
+
+if [ -z "${DATABRICKS_CLUSTER_HOSTNAME}" ]; then
+  echo "environment variable 'DATABRICKS_CLUSTER_HOSTNAME' must be set, exiting."
+  exit 1
+fi
+
+if [ -z "${SPLUNK_ACCESS_TOKEN}" ]; then
+  echo "environment variable 'SPLUNK_ACCESS_TOKEN' must be set, exiting."
+  exit 1
+fi
+
+if [ -z "${DATABRICKS_ACCESS_TOKEN}" ]; then
+  echo "environment variable 'DATABRICKS_ACCESS_TOKEN' must be set, exiting."
+  exit 1
+fi
+
+SPLUNK_OTEL_VERSION=${SPLUNK_OTEL_VERSION:-latest}
+OS="linux_amd64"
+SPLUNK_OTEL_BINARY_NAME="splunk_otel_collector"
+SPLUNK_OTEL_DOWNLOAD_BASE_URL="https://github.com/signalfx/splunk-otel-collector/releases"
+SPLUNK_OTEL_API_URL="https://api.github.com/repos/signalfx/splunk-otel-collector/releases/latest"
+SCRIPT_DIR=${SCRIPT_DIR:-/tmp/collector_download}
+CONFIG_FILENAME="config.yaml"
+SPLUNK_OTEL_BINARY_FILE="$SCRIPT_DIR/$SPLUNK_OTEL_BINARY_NAME"
+CONFIG_FILE="$SCRIPT_DIR/$CONFIG_FILENAME"
+SERVICE_PATH="/etc/systemd/system/"
+SERVICE_FILE="$SERVICE_PATH/$SPLUNK_OTEL_BINARY_NAME.service"
+
+if [ $SPLUNK_OTEL_VERSION = "latest" ]; then
+        SPLUNK_OTEL_VERSION=$(curl --silent "$SPLUNK_OTEL_API_URL" |    # Get latest Collector release from GitHub api
+            grep '"tag_name":' |                          # Get tag name line
+            sed -E 's/.*"([^"]+)".*/\1/')                 # Pluck latest release version
+        if [ -z "$SPLUNK_OTEL_VERSION" ]; then
+            echo "Failed to get tag_name for latest release from $SPLUNK_OTEL_VERSION/latest" >&2
+            exit 1
+        fi
+fi
+
+SPLUNK_OTEL_BINARY_DOWNLOAD_URL="${SPLUNK_OTEL_DOWNLOAD_BASE_URL}/download/${SPLUNK_OTEL_VERSION}/otelcol_${OS}"
+mkdir -p "$SCRIPT_DIR"
+INSTALLED_SPLUNK_OTEL_VERSION=""
+
+if [ -f $SPLUNK_OTEL_BINARY_FILE ]; then
+  # Output of `otelcol --version` is of the form:
+  # otelcol version vX.X.X
+  # Capture output of `otelcol --version` as an array, then extract version
+  INSTALLED_SPLUNK_OTEL_VERSION=($($SPLUNK_OTEL_BINARY_FILE --version))
+  INSTALLED_SPLUNK_OTEL_VERSION=${INSTALLED_SPLUNK_OTEL_VERSION[2]}
+fi
+
+if [ ! -f $SPLUNK_OTEL_BINARY_FILE ] || [ $INSTALLED_SPLUNK_OTEL_VERSION != $SPLUNK_OTEL_VERSION ]; then
+  # If the binary is already installed it's the wrong version and it needs to be removed before downloading again
+  # to avoid "Text file busy" errors
+  if [ -f $SPLUNK_OTEL_BINARY_FILE ]; then
+    rm "$SPLUNK_OTEL_BINARY_FILE"
+  fi
```

Comment:
After some more testing, I had to add this section to ensure the new version of the collector would be properly installed over an existing different collector version.